src/cdk-experimental/listbox/listbox.ts

```diff
@@ -6,26 +6,112 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Directive} from '@angular/core';
+import {
+  ContentChildren,
+  Directive,
+  ElementRef,
+  Input,
+  QueryList
+} from '@angular/core';
 
+/**
+ * Directive that applies interaction patterns to an element following the aria role of option.
+ * Typically meant to be placed inside a listbox. Logic handling selection, disabled state, and
+ * value is built in.
+ */
 @Directive({
   selector: '[cdkOption]',
   exportAs: 'cdkOption',
   host: {
     role: 'option',
+    '[attr.aria-selected]': '_selected',
+    '[attr.data-optionid]': '_optionId',
```

Comment:
It was my understanding that attr.id would overlap with the id that users can write in the templates. I wanted to make a new optionid field that users would not have set already. I used `data-optionid` because I had seen that format before, though I'm not sure where now. Perhaps just `optionid` would be best.